scripts/fetch_artsy.py

```diff
@@ -1,49 +1,58 @@
-import requests
-from bs4 import BeautifulSoup, SoupStrainer
-import time
-
-Name = 'Analogue'
-client_id = '77e9422bdf38d3bc0179'
-client_secret = '7eec7ad1ebb0b01272ae4eec0e71170a'
-starting_url = 'https://api.artsy.net/api'
-
-
-def fetch_artsy(url):
-    #authentication
-    artsy_id = url.split('/')[-1]
-    artsy_type = url.split('/')[-2]
-    # print(artsy_type, artsy_id)
-
-    api_data = {}
-    auth_url = '/tokens/xapp_token'
-    payload = {"client_id" : '77e9422bdf38d3bc0179',
-    "client_secret" : '7eec7ad1ebb0b01272ae4eec0e71170a'}
-    auth = requests.post(starting_url+auth_url, data=payload).json()
-    # token = auth['token']
-    
-    payload = {'X-Xapp-Token' : auth['token'], 'Content-Type': 'application/json', "client_id" : '77e9422bdf38d3bc0179',
-    "client_secret" : '7eec7ad1ebb0b01272ae4eec0e71170a'}
-    # newreq = requests.post(starting_url+auth_url, data=payload).json()
-    # print(newreq)
-
-    if artsy_type == 'artists': #
-        pass
-    elif artsy_type == 'artworks': #
-        pass
-    elif artsy_type == 'collections': #
-        pass
-    elif artsy_type == 'images':
-        pass
-    elif artsy_type == 'profiles':
-        pass
-    elif artsy_type == 'sale_artworks':
-        pass
-    elif artsy_type == 'sales':
-        pass
-    elif artsy_type == 'search': #
-        pass
-    elif artsy_type == 'shows': #
-        pass
-    else:
-        return api_data
+import requests
+
+
+def fetch_artsy(url):
+
+    client_id = '77e9422bdf38d3bc0179'
+    client_secret = '7eec7ad1ebb0b01272ae4eec0e71170a'
+    starting_url = 'https://api.artsy.net/api'    
+    
+    
+    artsy_id = url.split('/')[-1]
+    artsy_type = url.split('/')[-2]
+    
+    api_data = {}
+    #authentication
+    auth_url = '/tokens/xapp_token'
+    payload = {"client_id" : client_id,
+    "client_secret" : client_secret}
+    auth = requests.post(starting_url+auth_url, data=payload).json() #getting token
+    
+    headers = {
+        'accept': 'application/json',
+        'X-Xapp-Token': auth['token']
+    }
+
+    if artsy_type == 'artist':
+        target = starting_url+'/artists/'+artsy_id
+        api_data = requests.get(target, headers=headers).json()
+        try:
+            api_data.update(requests.get(api_data['_links']['artworks']['href'], headers=headers).json())
+        except:
+            pass
+    
+    elif artsy_type == 'artwork':
+        target = starting_url + '/artworks/' + artsy_id
+        api_data = requests.get(target, headers = headers).json()
+        try:
+            api_data.update(requests.get(api_data['_links']['artists']['href'], headers=headers).json())
+        except:

```

Comment:
we should have a log here saying when the api files, using `pass` is definitely a bad pattern

@joelrojo do we have any observability setup up to help the young bulls monitor things when they 🚢

Reply:
@hxghhhh not yet, but @FaazAbidi is setting up sentry so should have some basic optics into errors at least. You have any recommendations on what to integrate?